pkg/service/dependencies.go

```diff
@@ -38,6 +42,7 @@ func NewDependencies() (*Dependencies, error) {
 	if deps.Storage == nil || deps.Manipulator == nil {
 		return nil, errors.New("handler dependencies are not valid")
 	}
+	deps.MetricService = metricService
```

Comment:
This can be removed I think, once you set the prometheus http handler upon creation.
On second thought, you already have prometheus registry outside the `prometheusService`, you can set the handler outside this service.